test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1435 @@
-        commands_results = [
-            (0, 'Chain INPUT (policy DROP)\n', ''),
-            (0, '', '')
-        ]
-
-        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
-            run_command.side_effect = commands_results
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertTrue(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 2)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
-        self.assertEqual(run_command.call_args_list[1][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-P',
-            'INPUT',
-            'ACCEPT',
-        ])
-
-    def test_policy_table_no_change(self):
-        """Test don't change policy of a chain if the policy is right"""
-        set_module_args({
-            'policy': 'ACCEPT',
-            'chain': 'INPUT',
-        })
-        commands_results = [
-            (0, 'Chain INPUT (policy ACCEPT)\n', ''),
-            (0, '', '')
-        ]
-
-        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
-            run_command.side_effect = commands_results
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertFalse(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 1)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
-
-    def test_policy_table_changed_false(self):
-        """Test flush without parameters and change == false"""
-        set_module_args({
-            'policy': 'ACCEPT',
-            'chain': 'INPUT',
-            '_ansible_check_mode': True,
-        })
-        commands_results = [
-            (0, 'Chain INPUT (policy DROP)\n', ''),
-        ]
-
-        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
-            run_command.side_effect = commands_results
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertTrue(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 1)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
+    """Return iptables binary path."""
+    return IPTABLES_CMD
+
+
+@pytest.fixture
+def _mock_basic_commands(mocker):
+    """Mock basic commands like get_bin_path and get_iptables_version."""
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.get_bin_path",
+        return_value=get_bin_path(),
```

Comment:
I only now realized that the helper function is not actually needed.
```suggestion
        return_value=IPTABLES_CMD,
```